v3/reutlingen.py

```diff
@@ -0,0 +1,122 @@
+"""
+Copyright 2024 binary butterfly GmbH
+Use of this source code is governed by an MIT-style license that can be found in the LICENSE.txt.
+"""
+
+import csv
+import re
+from datetime import datetime, timezone
+from decimal import Decimal
+from enum import Enum
+from io import StringIO
+from typing import Any
+
+import pyproj
+from validataclass.dataclasses import validataclass
+from validataclass.exceptions import ValidationError
+from validataclass.validators import (
+    DataclassValidator,
+    EnumValidator,
+    IntegerValidator,
+    ListValidator,
+    NumericValidator,
+    StringValidator,
+)
+
+from common.base_converter import CsvConverter
+from common.exceptions import ImportParkingSiteException
+from common.models import ImportSourceResult
+from common.validators import StaticParkingSiteInput
+from common.validators.base_validators import ParkingSiteTypeInput
+from common.validators.fields.noneable import Noneable
+from util import SourceInfo
+
+
+class ReutlingenParkingSiteType(Enum):
+    PARKHAUS = 'parkhaus'
+    TIEFGARAGE = 'tiefgarage'
+    PARKFLAECHE = 'parkfläche'
+    P_R = 'p+r'
+
+    def to_parking_site_type_input(self) -> ParkingSiteTypeInput:
+        # TODO: find out more details about this enumeration for a proper mapping
+        return {
+            self.PARKHAUS: ParkingSiteTypeInput.CAR_PARK,
+            self.TIEFGARAGE: ParkingSiteTypeInput.UNDERGROUND,
+            self.PARKFLAECHE: ParkingSiteTypeInput.OFF_STREET_PARKING_GROUND,
+        }.get(self, ParkingSiteTypeInput.OTHER)
+
+
+class PointCoordinateTupleValidator(ListValidator):
+    PATTERN = r'POINT \(([-+]?\d+\.\d+) ([-+]?\d+\.\d+)\)'
+
+    def validate(self, input_data: Any, **kwargs) -> list:
+        input_match = re.match(self.PATTERN, input_data)
+        if input_match is None:
+            raise ValidationError(code='invalid_tuple_input', reason='invalid point coordinate tuple input')
+        input_data = [float(input_match.group(1)), float(input_match.group(2))]
+
+        return super().validate(input_data, **kwargs)
+
+
+@validataclass
+class ReutlingenRowInput:
+    uid: int = IntegerValidator(allow_strings=True)
+    type: ReutlingenParkingSiteType = EnumValidator(ReutlingenParkingSiteType)
+    coordinates: list = PointCoordinateTupleValidator(NumericValidator(), min_length=2, max_length=2)
+    capacity: str = Noneable(StringValidator())
```

Comment:
You do want an integer afterwards, and you can do that directly: just use the `ExcelNoneable` instead of `Nonable`, which will accept emptystring as an None value. Then the inner validator can be an `IntegerValidator` with strings allowed.